src/app/modules/form/public-form/public-form.controller.ts

```diff
@@ -161,3 +173,113 @@ export const handleRedirect: RequestHandler<
+    FormService.checkFormSubmissionLimitAndDeactivateForm(form),
+  )
+
+  // Early return if form is not public or any error occurred.
+  if (formResult.isErr()) {
+    const { error } = formResult
+    logger.error({
+      message: 'Error retrieving public form',
+      meta: {
+        action: 'handleGetPublicForm',
+        ...createReqMeta(req),
+        formId,
+      },
+      error,
+    })
+    const { errorMessage, statusCode } = mapRouteError(error)
+    return res.status(statusCode).json({ message: errorMessage })
+  }
+
+  const form = formResult.value
+  const publicForm = form.getPublicView()
+  const { authType } = form
+  let myInfoError = false
+
+  // NOTE: Creating a variable to ensure that TS will enforce the type and ensure all keys in AuthType are covered.
+  const formController: FormController<
+    | ReturnType<typeof SpcpFactory.createFormWithSpcpSession>
+    | ReturnType<typeof MyInfoFactory.createFormWithMyInfo>
+  > = {
+    [AuthType.SP]: () =>
+      SpcpFactory.createFormWithSpcpSession(form, req.cookies),
+    [AuthType.CP]: () =>
+      SpcpFactory.createFormWithSpcpSession(form, req.cookies),
+    [AuthType.MyInfo]: () => {
+      return MyInfoFactory.createFormWithMyInfo(form, req.cookies)
+        .andThen((publicForm) => {
+          return extractSuccessfulMyInfoCookie(req.cookies).map(
+            (myInfoCookie) => {
+              const cookiePayload: MyInfoCookiePayload = {
+                ...myInfoCookie,
+                usedCount: myInfoCookie.usedCount + 1,
+              }
+              // NOTE: This is a side effect to set the cookie on the result after it has been successfully prefilled.
+              res.cookie(
+                MYINFO_COOKIE_NAME,
+                cookiePayload,
+                MYINFO_COOKIE_OPTIONS,
+              )
+              return publicForm
+            },
+          )
+        })
+        .mapErr((error) => {
+          // NOTE: This is a side-effect as there is no need for cookie if data could not be retrieved
+          res.clearCookie(MYINFO_COOKIE_NAME, MYINFO_COOKIE_OPTIONS)
+          // NOTE: This is done as a workaround because the type of FormController enforces a uniform return value
+          // but we are required to signal if myInfoError in the event of an error
+          myInfoError = true
+          return error
+        })
+    },
+    [AuthType.NIL]: () => okAsync({ form: form.getPublicView() }),
+  }
+
+  // NOTE: Once there is a valid form retrieved from the database,
+  // the client should always get a 200 response with the form's public view.
+  // Additional errors should be tagged onto the response object like myInfoError.
+  return formController[authType]()
+    .andThen((publicFormView) =>
+      FormService.setIsIntranetFormAccess(getRequestIp(req), publicFormView),
+    )
+    .andThen((publicFormView) =>
+      FormService.setMyInfoError(publicFormView, myInfoError),
```

Comment:
As stated above, `myInfoError` is **guaranteed** to be `false` here and this step does not actually need to be called.

We should remove this step to maintain parity to the results prior to the refactor;
- when there are no MyInfo errors, `myInfoError` key should not be sent back to the client.

Reply:
unclear to me whether there's any advantage in letting a key be undefined, but will defer to @karrui

Reply:
There is right, if a key has an undefined value when it serializes it will not even appear on the client.

```ts
JSON.parse(JSON.stringify({someKey: undefined})) // {}
```

Reply:
so i think the main advantage here would be that we prevent any funny stuff on happening with `undefined` - if we do adopt this approach, the type is `boolean | undefined` instead of just `boolean`.

Reply:
> There is right, if a key has an undefined value when it serializes it will not even appear on the client.

how is this an advantage though? as @seaerchin said above, isn't it better to just return `boolean` instead of `boolean | undefined`?

anw I don't have a super strong opinion on this la so again will defer to @karrui, I'm fine with keeping existing behaviour as well

Reply:
we can tighten the return type to `true | undefined`? If it exists the key will be there, otherwise it won't. I am just mindful about returning flags that 99% of the time is not useful

Reply:
another way of looking at this: maybe @seaerchin can make a decision on whether returning `myInfoError: false` would make the code cleaner. in that case I think it's worth changing, but otherwise we should stick to `undefined`. rn it seems we're jumping through a lot of hoops just to be able to return `myInfoError: false` in every case, which is not necessary.

Reply:
sure I'm fine with `true | undefined`

Reply:
going with `true | undefined`